aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -56,6 +56,27 @@ Tensor& bitwise_not_out(Tensor& result, const Tensor& self) {
   return result;
 }
 
+Tensor logical_not(const Tensor& self) {
+  Tensor result = at::empty({0}, self.options());
+  return at::logical_not_out(result, self);
+}
+
+Tensor& logical_not_(Tensor& self) {
+  return at::logical_not_out(self, self);
+}
+
+Tensor& logical_not_out(Tensor& result, const Tensor& self) {
+  TORCH_CHECK(self.scalar_type() == kBool,
+              "logical_not only supports bool tensors.");
```

Comment:
nit: I'd say "only currently supports" to make it clear we are planning to add support for other types.